### PR TITLE
Make barcode sync finite

### LINKED ISSUE #1748: Infinite barcode sync
## What went wrong? 😲

When barcode is created (probably when updated too), it causes infinite sync, this was noted here: https://github.com/openmsupply/open-msupply/pull/1725#pullrequestreview-1419705472

## Expected behaviour 🤔

Does not cause infinite sync, see proposed solution

## How to Reproduce 🔨

[Using this data file](https://drive.google.com/drive/u/1/search?q=short%20configure)

[And this branch](https://github.com/openmsupply/open-msupply/pull/1725#discussion_r1191789651) with [this change](https://github.com/openmsupply/open-msupply/files/11457192/patch.git.zip)(change is required, based on this comment):  

Start two seperate omSupply instances:

And sync them (credentials = supplier->pass, test->pass):

`APP_SERVER__PORT=8000 APP_DATABASE__DATABASE_NAME=db1name cargo run`
and in another terminal
`APP_SERVER__PORT=9000 APP_DATABASE__DATABASE_NAME=db2name cargo run `

On the supplier one do this:

```graphql
mutation MyMutation {
  updateStockLine(
    input: {id: "cce96ae0-72c2-4332-877d-47f55b6149c7", barcode: "somethingorother"}
    storeId: "0FD73FA090764B40AEFCAE38D4FA4012"
  ) {
    ... on StockLineNode {
      id
      availableNumberOfPacks
      barcode
    }
  }
}
```

## Other comments

I think we should have caught this in: https://github.com/openmsupply/open-msupply/pull/1555, it definitely needs more documentation but the reviewers should be aware now different sync data types should be implemented (i.e. if there is pull and push without store_id on the record, it's probably a shared type, which needs is_sync_update, this was introduced in program works originally)

## Proposed solution

is_sync_update, as done for requisitions here:

https://github.com/openmsupply/open-msupply/pull/1721

And database changes here: https://github.com/openmsupply/open-msupply/commit/5b41b5bdfddcbbca42edb898f2ca27a118c9f0db

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1748 

# 👩🏻‍💻 What does this PR do? 
 
Implement the `is_sync_update` for barcodes.
Also renames the `value` column to `gtin`

# 🧪 How has/should this change been tested? 

As per the issue - running two sites from the same datafile. Create a barcode and sync it around.

## 📃 Documentation

_No user facing changes_

### CODE REVIEW

File: server/repository/src/migrations/v1_01_12/barcode_sync.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        ALTER TABLE barcode ADD is_sync_update bool NOT NULL DEFAULT False;
+        ALTER TABLE public.barcode RENAME COLUMN value TO gtin;
```
Comment (andreievg) [inline server/repository/src/migrations/v1_01_12/barcode_sync.rs:8]: We also need to re-juggle the triggers, as per commit mentioned in issue: https://github.com/openmsupply/open-msupply/commit/5b41b5bdfddcbbca42edb898f2ca27a118c9f0db

**Review by andreievg — CHANGES_REQUESTED** [review]
Doesn't compile, also please see change request.

Can I pretty please ask to break down PRs to solve their own respective issues, it makes Reviewer job a lot easier, especially when there is a lot of pressure on reviewer to look at urgent PRs, it also allows to merge seperate issues faster.

I've ended up making an issues in the PR that i reviewed (was hoping an issue would be made automatically based on comment, I did see you added a point to is_sync_update issue, but it's really hard to see for me, unless it's a separate issue that is linked to original request), i ended up creating the issue myself btw, and linked it to that [comment](https://github.com/openmsupply/open-msupply/pull/1725#discussion_r1193226909)

File: server/repository/src/migrations/v1_01_12/barcode_sync.rs (hunk 1 above)
Comment (mark-prins) [reply server/repository/src/migrations/v1_01_12/barcode_sync.rs:8]: like this? 69f6c510c

Comment (andreievg) [reply server/repository/src/migrations/v1_01_12/barcode_sync.rs:8]: 🙏

Just doing cargo test and manual test

**Comment by mark-prins** [comment]
Test compilation fixed in ec90fd437

**Review by andreievg — APPROVED** [review]
Thank you @mark-prins, should be good after the change to `{operation}` in sqlite migrations  `AFTER {operation}`

**Pull request merged**